src/Symfony/Component/Form/Extension/Core/Type/TextType.php

```diff
@@ -30,13 +32,24 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         }
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        if (isset($options['placeholder']) && '' != $options['placeholder']) {
+            $view->vars['attr']['placeholder'] = $options['placeholder'];
+        }
+    }
+
     /**
      * {@inheritdoc}
      */
     public function configureOptions(OptionsResolver $resolver)
     {
         $resolver->setDefaults([
             'compound' => false,
+            'placeholder' => '',
```

Comment:
You can restrict type with `$resolver->setAllowedTypes('placeholder', 'string');`